### ISSUE TITLE
Polybar can't compile due to issue with upstream curl

### ISSUE THREAD

**Issue by intrk** [jaagr/polybar#647]
[ 75%] Building CXX object bin/CMakeFiles/polybar.dir/utils/http.cpp.o
/home/gregf/.cache/pacaur/polybar-git/src/polybar/src/utils/http.cpp:2:10: fatal error: 'curl/curlbuild.h' file not found
#include <curl/curlbuild.h>
^~~~~~~~~~~~~~~~~~
[ 76%] Building CXX object bin/CMakeFiles/polybar.dir/utils/i3.cpp.o
1 error generated.
make[2]: *** [bin/CMakeFiles/polybar.dir/build.make:1359: bin/CMakeFiles/polybar.dir/utils/http.cpp.o] Error 1

Seems like <curl/buildcurl.h> has been removed from upstream and this is breaking the compilation. 
I tried removing the offending include statement and compilation was successful.

**Comment by patrick96** [comment]
Thanks for reporting. The curlbuild file was removed in commit https://github.com/curl/curl/commit/73a2fcea0b4adea6ba342cd7ed1149782c214ae3 which is not part of any release yet.
I will submit a PR to remove the include since on my system which still has the `curl/curlbuild.h` file it compiles fine without it.

**Comment by intrk** [comment]
I came across this from the Arch AUR, how do I go about requesting that package also be updated?

**Comment by patrick96** [comment]
Which package do you want updated?

**Comment by intrk** [comment]
https://aur.archlinux.org/packages/polybar/ ? 
New to this stuff so sorry for my ignorance.

**Comment by patrick96** [comment]
The package you linked is currently on the newest version of polybar.
If you want to always use the most up date commits, install https://aur.archlinux.org/packages/polybar-git

If you're asking how you can request that the bug with the `curl/curlbuild.h` is fixed, then posting it here was the right thing to do. I have already fixed it and sent it to jaagr to merge into this repo (see #648). Once it is merged, you can update `polybar-git` and it will compile successfully

**Comment by intrk** [comment]
Awesome, thanks mate! :)

**Comment by yorickvP** [comment]
There should probably be a new release with this in, it started failing to build on nixos.

**Comment by patrick96** [comment]
Yes there should, but currently the master branch has some issues and also I think jaagr is really busy at the moment

**Comment by hubbcaps** [comment]
any possibility of getting traction on this on master? This is still failing on gentoo under the only up to date portage overlay at the moment, having to revert to an unlisted, unofficial overlay to get an older working build currently.

**Comment by yorickvP** [comment]
@patrick96 maybe make a release branch?

**Comment by patrick96** [comment]
@hubbcaps The fix for the build issue has been in master for a few months already.
If you are building against the `3.0.5` tag, then you can cherry pick the commit that fixes it before building:
```bash
git cherry-pick -n d35abc7620c8f06618b4708d9a969dfa2f309e96
```
